### PR TITLE
docs(combo): add combo valueKey binding documentation

### LINKED ISSUE #1479: Add combo breaking change to Update guide
Add the [following ](https://github.com/IgniteUI/igniteui-angular/pull/5569/files) combo breaking change to Update guide.

### LINKED ISSUE #1480: Improve Combo Value Binding paragraph
Update Combo Value Binding [paragraph](https://www.infragistics.com/products/ignite-ui-angular/angular/components/combo_features.html#value-binding), using the new combo [improvement](https://github.com/IgniteUI/igniteui-angular/issues/5217).

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Closes #1479 
Closes #1480 

Samples are added in https://github.com/IgniteUI/igniteui-angular-samples/pull/1449

### CODE REVIEW

File: en/components/combo_features.md [hunk 1]
```diff
@@ -87,27 +87,134 @@ export class ComboDemo implements OnInit {
...
+ - [`selectItems`]({environment:angularApiUrl}/classes/igxcombocomponent.html#selectitems), [`deselectItems`]({environment:angularApiUrl}/classes/igxcombocomponent.html#deselectitems), [`selectAllItems`]({environment:angularApiUrl}/classes/igxcombocomponent.html#selectallitems), [`deselectAllItems`]({environment:angularApiUrl}/classes/igxcombocomponent.html#deselectallitems) - these methods should be called with the corresponding `valueKey` property values if `[valueKey]` is specified for the combo.
+ - [`onSelectionChange`]({environment:angularApiUrl}/classes/igxcombocomponent.html#onselectionchange) - the `newSelection`, `oldSelection`, `added` and `removed` arrays of the [`IComboSelectionChangeEventArgs`]({environment:angularApiUrl}/interfaces/icomboselectionchangeeventargs.html) will hold the corresponding values of the items. 
+
+### Selection without valueKey
+When no `[valueKey]` is provided, the combo handles selection based on **equality (===)**.
+If the data is of a primitive type (e.g. `string`, `number`), the provided array to selection methods should contain only the values that you'd like to select. For example:
```
Comment (dafo) [inline en/components/combo_features.md]: I think "you would" be more suitable. If you decide so - change this everywhere in the topic.

File: en/components/combo_features.md [hunk 2]
```diff
@@ -87,27 +87,134 @@ export class ComboDemo implements OnInit {
...
 
 
 Follow this [topic](combo_remote.md) for more details about binding [igx-combo]({environment:angularApiUrl}/classes/igxcombocomponent.html) with remote data.
 
+
+## Selection
+The combo handles selection differently, depending on its [valueKey]({environment:angularApiUrl}/classes/igxcombocomponent.html#valuekey) property.
+
+### Selection with valueKey
+If a `[valueKey]` is provided, all of the combo's selection events will be handled with the value of the items' `valueKey` property.
```
Comment (dafo) [inline en/components/combo_features.md]: If a `valueKey` is provided

File: en/components/combo_remote.md [hunk 3]
```diff
@@ -126,10 +130,18 @@ export class ComboRemoteComponent implements OnInit {
...
+
+#### Handling Selection
+When using a combo bound to remote data, loaded in chunks, dealing with a more complex data type (e.g. objects), it is necessary to define a `valueKey`. As stated in the [selection topic](../combo_features.md#selection), when no `valueKey` is specified, the combo will try to handle selection by **equality (===)**. Since the objects that will be marked as selected will **not be the same** as the object that are continuously loaded, the selection will fail.
+When binding the combo to remote data, make sure to specify a `valueKey`, representing a property that is unique to each item.
 
-> Note: Anytime new data is loaded, we update `totalItemCount` [igx-combo]({environment:angularApiUrl}/classes/igxcombocomponent.html) property, in order to have proper size of the list's scroll bar. In that case the service returns total size using the property `@odata.count`.
+When the combo is bound to remote data, setting value/selected items through API will selection methods will only take into account the items that are **loaded in the current chunk**. If you set an initial value, make sure those specific items are loaded before selecting.
```
Comment (wnvko) [inline en/components/combo_remote.md]: ...  through API will selection methods will ...
this is leftover maybe

File: en/components/combo_features.md (hunk 2 above)
Comment (wnvko) [reply en/components/combo_features.md]: as you changed [`valueKey`] to `valueKey` in the sample markup you should change it everywhere in the text

File: en/components/general/update_guide.md [hunk 4]
```diff
@@ -44,6 +44,13 @@ Unfortunately not all changes can be automatically updated. Changes bellow are s
 
 For example: if you are updating from version 6.2.4 to 7.1.0 you'd start from the "From 6.x .." section apply those changes and work your way up:
 
+### From 8.1.x to 8.2.x
+* The way that the [`igx-comb`](../combo.md) handles selection and data binding is changed.
+ - If the combo's [`valueKey`] input is defined, the control will look for that specific property in the passed array of data items when performing selection.
```
Comment (wnvko) [inline en/components/general/update_guide.md]: [`valueKey`] => `valueKey`

File: en/components/combo_features.md (hunk 2 above)
Comment (ViktorSlavov) [reply en/components/combo_features.md]: I wanted to accent that it was an input by bracing it w/ `[`, but apparently it backfired 🐸 . Thanks for pointing it out, I'll remove the brackets in the text blocks.

**Pull request merged**